cinq_auditor_required_tags/providers.py

```diff
@@ -125,11 +125,49 @@ def terminate_ec2_instance(client, resource):
             error
         ))
 
+def delete_s3_bucket(client, resource):
+    try:
+        session = get_aws_session(resource.account)
+        bucket = session.resource('s3', resource.location).Bucket(resource.resource_id)
+
+        lifecyclePolicy = {'Rules': [
+            {'Status': 'Enabled',
+                'NoncurrentVersionExpiration': {u'NoncurrentDays': 1},
+                'Filter': {u'Prefix': ''},
+                'Expiration': {u'Days': 1},
+                'AbortIncompleteMultipartUpload': {u'DaysAfterInitiation': 1},
+                'ID': 'cloudInquisitor'}]}
+
+        objects = list(bucket.objects.limit(count=1))
+        versions = list(bucket.object_versions.limit(count=1))
+        if not objects and not versions:
```

Comment:
If you are only testing for presence does it matter if these are lists?

Reply:
`bucket.objects.all()` and `bucket.objects.limit` both return iterators (despite limiting at one object). calling `list(iterator)` will iterate through it and return it as a list, which makes it easier to test for presence than writing a for loop and iterating on our own.

the goal with those is to make sure the bucket is actually empty, if there's at least one object or at least one version in there the call to `bucket.delete()` will throw an exception